updates to MYNN-EDMF and Thompson

joeolson42 opened the pull request:
The following PR consists of updates to 3 schemes and related modules:

1. MYNN-EDMF

- optimization work: fixed 8-plume model instead of variable number plume. maintains performance
- bug fix to correct grid mean vs in-cloud mixing ratios. Now requires namelist variable lcnorm (set to true)
- adjustments to cloud pdf to better fit modified thompson scheme

2. MYNN surface layer

- tweak to Czil parameter to improve diurnal cycle of T and wsp.

3. Thompson aerosol-aware microphysics

- change to nucleation rates to fix marine drizzle problem
- minimum effective radii of 7 microns over water

tanyasmirnova committed:
Changes in RUC LSM:
1. adding a heat source from fires. It is icontrolled by a flag and not turned on yet. Needs more testing.
2. Adding output of surface paramteres: leaf area index (LAI), wilting point and soil field capacity.

tanyasmirnova committed:
Fire heat flux computation is added to the smoke-dust code.

tanyasmirnova committed:
Fix the segmentation fault error in computing fire heat flux.

joeolson42 committed:
updates from HFIP dev branch

joeolson42 commented:
Here's some supporting documentation for this PR:
[Overview of Updates to MYNN_ and Thompson.pdf](https://github.com/ufs-community/ccpp-physics/files/12929695/Overview.of.Updates.to.MYNN_.and.Thompson.pdf)

joeolson42 committed:
missed a variable in the argument for activ_ncloud during the intial merge.

grantfirl commented on physics/module_bl_mynn.F90:
```diff
@@ -1,5 +1,5 @@
 !>\file module_bl_mynn.F90
-!! This file contains the entity of MYNN-EDMF PBL scheme.
+! This file contains the entity of MYNN-EDMF PBL scheme.
```
Just to be sure, was the removal of the second `!` intentional? Its removal means that Doxygen will not parse this line anymore. Not that the file description is all that useful in this case (and `entity` should probably be `entirety`), but still.

joeolson42 replied:
Inadvertent. I'll add it back.

tanyasmirnova committed:
Addressed the reviewer's comments.

tanyasmirnova committed:
Changes in module_sf_ruclsm.F90:
1. Removed commented lines in fire feedback to RUC LSM.
2. The coordinates of a test point are passed to the RUC LSM subroutines to make debug printing more manageable.
3. Reverted change in the TRANSF subroutine to original version. The non-linear root distribution needs scientific evaluation in the retro runs.

grantfirl commented on physics/module_bl_mynn.F90:
```diff
@@ -7066,14 +7087,14 @@ SUBROUTINE DDMF_JPL(kts,kte,dt,zw,dz,p,              &
-            ENT(k,i) = min(ENT(k,i),0.9/(ZW(k+1)-ZW(k)))
-         enddo
-      enddo
+!      ! entrainent: Ent=Ent0/dz*P(dz/L0)
```
Consider removing the commented out code and relying on git history to recover these statements if it is every necessary to.

joeolson42 replied:
I probably will remove it at some point. That subroutine is under construction. It will be gone by the next PR.

grantfirl reviewed (approved):
Looks fine from CCPP point of view.

tanyasmirnova committed:
Merge remote-tracking branch 'upstream/ufs/dev' into ruclsm_lai

joeolson42 committed:
Adding back a missing exclamation point in module_bl_mynn.F90 and adding a limit to protect against the ridiculously large z0s when using thin first model levels in module_sf_mynn.F90

tanyasmirnova committed:
Based on 1-D testing in ESM-SnowMIP a change to computation of snow thermal conductivity is made in module_sf_ruclsm.F90.

tanyasmirnova committed:
In module_sf_ruclsm.F90: clean-up comments, changed the constant for hard snow slab thermal conductivity.

joeolson42 committed:
Further tweaks to improve the MYNN, will update the attached pdf.

AndersJensen-NOAA committed:
Hail diagnostic

AndersJensen-NOAA committed:
Max hail diameter

dustinswales commented on physics/module_bl_mynn.F90:
```diff
@@ -3042,7 +3045,8 @@ SUBROUTINE  mym_turbulence (                                &
        pdk(k) = elq*( sm(k)*gm(k)                &
             &        +sh(k)*gh(k)+gamv ) +       &
-            &   TKEprodTD(k)
+!fix?            &   TKEprodTD(k)
```
Please remove

joeolson42 replied:
removed

dustinswales commented on physics/module_bl_mynn.F90:
```diff
@@ -3086,9 +3090,10 @@ SUBROUTINE  mym_turbulence (                                &
        
        !! Buoyncy term takes the TKEprodTD(k) production now
-       qBUOY1D(k) = elq*(sh(k)*gh(k)+gamv)+TKEprodTD(k) !staggered
+!fix?  qBUOY1D(k) = elq*(sh(k)*gh(k)+gamv)+TKEprodTD(k) !staggered
```
ditto (Git history captures these changes. Of course, unless you have a reason for keeping these comments in here?)

joeolson42 replied:
removed

dustinswales commented on physics/module_bl_mynn.F90:
```diff
@@ -3794,29 +3800,31 @@ SUBROUTINE  mym_condensation (kts,kte,   &
         !Diagnostic statistical scheme of Chaboureau and Bechtold (2002), JAS
         !but with use of higher-order moments to estimate sigma
-        PBLH2=MAX(10.,PBLH1)
+        pblh2=MAX(10.,pblh1)
```
To avoid precision loss when calling intrinsics you should be explicit about the precision (e.g 10._kind_phys instead of 10.)

joeolson42 replied:
fixed

dustinswales commented on physics/module_bl_mynn.F90:
```diff
@@ -3794,29 +3800,31 @@ SUBROUTINE  mym_condensation (kts,kte,   &
+           t      = th(k)*exner(k)
+           xl     = xl_blend(t)              ! obtain latent heat
+           qsat_tk= qsat_blend(t,  p(k))     ! saturation water vapor mixing ratio at tk and p
+           rh(k)  = MAX(MIN(1.00,qw(k)/MAX(1.E-10,qsat_tk)),0.001)
```
Ditto on the precision.

joeolson42 replied:
fixed

dustinswales commented on physics/module_bl_mynn.F90:
```diff
@@ -3826,14 +3834,24 @@ SUBROUTINE  mym_condensation (kts,kte,   &
+           !sgm(k) = max( sgm(k), qsat_tk*0.035 )
+
+           !introduce vertical grid spacing dependence on min sgm
+           wt     = max(500. - max(dz(k)-100.,0.0), 0.0)/500. !=1 for dz < 100 m, =0 for dz > 600 m
```
precision.

joeolson42 replied:
fixed

dustinswales commented on physics/module_bl_mynn.F90:
```diff
@@ -3850,6 +3868,14 @@ SUBROUTINE  mym_condensation (kts,kte,   &
            endif
+           !ensure adequate RH & q1 when qs is at least 1e-7 (above the PBLH)
+           if (qs(k)>1.e-7 .and. zagl .gt. pblh2) then
+              rh_hack =min(1.0, rhcrit + 0.08*(7.0 + log10(qs(k))))
```
precision.

joeolson42 replied:
fixed

dustinswales commented on physics/module_bl_mynn.F90:
```diff
@@ -3922,17 +3945,22 @@ SUBROUTINE  mym_condensation (kts,kte,   &
+              Fng = min(23.9 + exp(-1.6*(q1k+2.5)), 60.)
+           endif
+
+           cfmax = min(cldfra_bl1D(k), 0.6)
```
precision

joeolson42 replied:
fixed

dustinswales commented on physics/module_bl_mynn.F90:
```diff
@@ -5889,153 +5921,166 @@ SUBROUTINE DMP_mf(                            &
-  Psig_w = MIN(Psig_w, Psig_shcu)
-  !print*," maxw=", maxw," Psig_w=",Psig_w," Psig_shcu=",Psig_shcu
+  !do nothing for small w (< 1 m/s), but linearly taper off for w > 1.0 m/s
+  maxw = max(0.,maxw - 1.0)
```
precision

dustinswales commented on physics/module_bl_mynn.F90:
```diff
@@ -5889,153 +5921,166 @@ SUBROUTINE DMP_mf(                            &
   ! Criteria (3)
-    maxwidth = MIN(maxwidth,0.5*cloud_base)
+    if ((landsea-1.5) .lt. 0) then  !land
+       maxwidth = MIN(maxwidth, 0.5*cloud_base)
```
precision

joeolson42 replied:
fixed

dustinswales commented on physics/module_bl_mynn.F90:
```diff
@@ -5889,153 +5921,166 @@ SUBROUTINE DMP_mf(                            &
       !width_flx = MAX(MIN(1000.*(0.6*tanh((flt - 0.050)/0.03) + .5),1000.), 0.)
-      width_flx = MAX(MIN(1000.*(0.6*tanh((flt - 0.040)/0.03) + .5),1000.), 0.) 
+      !width_flx = MAX(MIN(1000.*(0.6*tanh((flt - 0.040)/0.03) + .5),1000.), 0.) 
+      width_flx = MAX(MIN(1000.*(0.6*tanh((fltv - 0.040)/0.04) + .5),1000.), 0.)
```
precision

joeolson42 replied:
fixed

dustinswales commented on physics/module_bl_mynn.F90:
```diff
@@ -5889,153 +5921,166 @@ SUBROUTINE DMP_mf(                            &
-       IF(I > NUP2) exit
-       l  = dl*I                            ! diameter of plume
+    d  =-1.9  !set d to value suggested by Neggers 2015 (JAMES).
+    dl = (maxwidth - minwidth)/real(nup-1)
```
When calling "real", you should provide the precision, so real(nup-1,kind=kind_phys)

joeolson42 replied:
fixed

dustinswales commented on physics/module_mp_thompson.F90:
```diff
@@ -5846,6 +5856,7 @@ subroutine calc_effectRad (t1d, p1d, qv1d, qc1d, nc1d, qi1d, ni1d, qs1d,   &
          endif
          lamc = (nc(k)*am_r*g_ratio(inu_c)/rc(k))**obmr
          re_qc1d(k) = SNGL(0.5D0 * DBLE(3.+inu_c)/lamc)
+         if (lsml .ne. 1) re_qc1d(k) = max(re_qc1d(k), 7.0E-6)
```
Note. There's no need to be explicit about the precision here, since the precision is not parameterized in this routine.

dustinswales commented on physics/module_sf_mynn.F90:
```diff
@@ -1380,6 +1380,8 @@ SUBROUTINE SFCLAY1D_mynn(flag_iter,                            &
           ZNTstoch_lnd(I)  = ZNT_lnd(I)
        endif
+       !add limit to prevent ridiculous values of z0 (more than dz/15)
+       ZNTstoch_lnd(I) = min(ZNTstoch_lnd(I), dz8w1d(i)*0.0666)
```
precision

joeolson42 replied:
fixed

dustinswales reviewed (changes requested):
Everything looks good, just my normal concerns about precision, which should get resolved (mainly for portability concerns)

AndersJensen-NOAA committed:
Fixes after debug

AndersJensen-NOAA committed:
real to double for graupel psd parameters

joeolson42 committed:
Updating snow cloud fractions in MYNN-EDMF and removing redundant logic in sgscloud_radpre.F90

joeolson42 committed:
Addressing comments from Dustin

joeolson42 committed:
Addressing comments from Dustin

joeolson42 committed:
Adding some comments to describe overall changes.

joeolson42 committed:
fix typo

joeolson42 committed:
Merge branch 'ufs/dev' into mynnthomp_oct2023

grantfirl committed:
Merge branch 'max_hail_mp' into mynnthomp_oct2023

grantfirl commented:
@dustinswales @Qingfu-Liu Please re-review when you get a chance since some more changes were merged in from @AndersJensen-NOAA (https://github.com/ufs-community/ccpp-physics/pull/120/commits/5351c1695b85ee1354e9792f6b155e7dc2c8f82b)

Qingfu-Liu commented on physics/module_mp_thompson.F90:
```diff
@@ -2464,17 +2469,17 @@ subroutine mp_thompson (qv1d, qc1d, qi1d, qr1d, qs1d, qg1d, ni1d,    &
+      ! enddo
+      call graupel_psd_parameters(kts, kte, rand1, rg, ilamg, N0_g)
       endif
 
```
The comment lines 2472-2481, 3549-3558, and 6104-6113 can be removed

RuiyuSun commented:
Interesting! @joeolson42. I conducted an experiment with rew_min = 7um and compared with previous experiments, p8 and HR1. Please see the [a summary here.](https://docs.google.com/presentation/d/1HvwN3_Fu14DZRgT7S5bSA2UrZAX2884I/edit#slide=id.p3) . Based on the limited test results (from 6 cases) increasing the minimum REW led to a large global averaged downward SW bias.

yangfanglin commented:
@joeolson42 @AndersJensen-NOAA It appears a lot of changes have been made to Thompson MP in this PR.  In addition to REW will the other modifications change result as well ?  If this PR  alter GFSv17 and GEFSv13 results, we need to do more testing and evaluation.

dustinswales commented on physics/module_mp_thompson_make_number_concentrations.F90:
```diff
@@ -35,7 +36,7 @@ elemental real function make_IceNumber (Q_ice, temp)
       REAL, PARAMETER:: Ice_density = 890.0
       !REAL, PARAMETER:: PI = 3.1415926536
-      real, intent(in):: Q_ice, temp
+      real(kind_phys), intent(in):: Q_ice, temp
```
Why introduce using mixed precision in this module?

AndersJensen-NOAA replied:
@joeolson42 feel free to roll this back. I talked with Dustin and will add this in a new PR. Ignore all changes to physics/module_mp_thompson_make_number_concentrations.F90

grantfirl replied:
@joeolson42 @AndersJensen-NOAA Would you like me to revert this change for final testing of this PR?

dustinswales commented on physics/mp_thompson.meta:
```diff
@@ -610,6 +610,14 @@
+  dimensions = (horizontal_loop_extent)
+  type = real
+  kind = kind_phys
+  intent = out
```
This is defined with intent=inout in the .F90 file.
Please clarify.

AndersJensen-NOAA replied:
@dustinswales @joeolson42 to be consistent with other diagnostics like refl_10cm, this should be defined here as inout.

grantfirl replied:
I'll change this to "inout".

joeolson42 commented:
@yangfanglin The other changes (outside of the effective radius limit) shouldn't impact the non-aerosol-aware version. There were some hail diagnostic changes that would impact the results, but not in a prognostic sense.

tanyasmirnova committed:
Removed requested by the reviewer commented lines.

joeolson42 committed:
Better range checks for output in surface layer, tweaked cloud fractions. last commit before reg tests

AndersJensen-NOAA commented on physics/module_mp_thompson.F90:
```diff
@@ -6471,6 +6488,86 @@ SUBROUTINE semi_lagrange_sedim(km,dzl,wwl,rql,precip,pfsan,dt,R1)
+      ygra1 = alog10(max(1.e-9, rg(k)))
+      zans1 = 3.4 + 2./7.*(ygra1+8.) + rand1
+      N0_exp = 10.**(zans1)
+      N0_exp = max1(dble(gonv_min), min(N0_exp, dble(gonv_max)))
```
@grantfirl

grantfirl replied:
@AndersJensen-NOAA We settled on changing this to "max" or "dmax1", right? "max1" should be returning an integer that is then cast to double precision, which I don't think is what you want, right?

grantfirl replied:
Nevermind, I see that you already did this and this is outdated.

AndersJensen-NOAA committed:
Updates to Thompson MP after code review

joeolson42 committed:
removing the effective radii limit due to concerns by GFS developers

joeolson42 committed:
Merge branch 'ufs/dev' into mynnthomp_oct2023

AndersJensen-NOAA committed:
Bug fix for divide by zero in hail size

grantfirl commented on physics/module_bl_mynn.F90:
```diff
@@ -3863,28 +3864,28 @@ SUBROUTINE  mym_condensation (kts,kte,   &
-           if (qi(k)>1.e-9) then
-              rh_hack =min(1.0_kind_phys, rhcrit + 0.07*(9.0 + log10(qi(k))))
+           !ensure adequate RH & q1 when qi is at least 1e-9 (above the PBLH)
+           if (qi(k)>1.e-9 .and. zagl .gt. pblh2) then
```
@joeolson42 This is the change currently causing the same failure as in https://github.com/ufs-community/ccpp-physics/issues/114.

joeolson42 replied:
I don't think there is anything wrong with that line of code. This just seems to be (one of many) tweaks that can apparently push the model over the edge. You aren't suggesting this is actually the culprit, right?

grantfirl replied:
Right, I'm just saying that this line is the "straw that breaks the camel's
back" for this PR with respect to the known issue, and removing this change
would cause all RTs to at least run to completion, which, as you know, is
an arbitrary bar. So, in that sense, this is the proximate cause and not
the ultimate cause.

gthompsnWRF commented:
@AndersJensen-NOAA  and @joeolson42   Is RRFS using `subroutine progcld_thompson` within `radiation_clouds.f`?

I am asking because I have concerns that the cloud fractions between zero and one might be given the value of `re_qc_min` (2.5 microns) versus what I assume for the "fake clouds" which is found in these lines of code . . .
```
            if (qc1d(k).gt.clwmin .and. cldfra1d(k).lt.ovcst) then
               cwp(i,k) = qc1d(k) * dz1d(k)*1000.
               if ((xland-1.5).GT.0.) then                               !--- Ocean
                  rew(i,k) = 9.5
               else                                                      !--- Land
                  rew(i,k) = 5.5
               endif
            endif
```

So I am assuming rather large droplets in the fractional clouds for which there could be zero grid-scale microphysics for which to compute the mathematical value of cloud water effective radii.  In that case, the subroutine within module_mp_thompson to compute effective radii would return `re_qc_min` (or 2.5 microns), but it can be overridden here in the case with cloud fraction less than 1.0 (but greater than zero) so the clouds won't be so bright.  Also, in your PDF, what are you meaning by clouds are "too bright?"  If you're looking at things relative to some pseudo satellite operator in the visible spectra, then who has done the checks for consistency with the operator calculating albedo or whatever?  Those include tables for single-scattering albedo and asymmetry factor and so on.

Also tagging @RuiyuSun and @yangfanglin

grantfirl commented:
Right, I'm just saying that this line is the "straw that breaks the camel's
back" for this PR with respect to the known issue, and removing this change
would cause all RTs to at least run to completion, which, as you know, is
an arbitrary bar. So, in that sense, this is the proximate cause and not
the ultimate cause.

On Wed, Nov 8, 2023 at 6:30 PM Joseph Olson ***@***.***>
wrote:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In physics/module_bl_mynn.F90
> <https://github.com/ufs-community/ccpp-physics/pull/120#discussion_r1387309169>
> :
>
> > @@ -3863,28 +3864,28 @@ SUBROUTINE  mym_condensation (kts,kte,   &
>             !Add condition for falling/settling into low-RH layers, so at least
>             !some cloud fraction is applied for all qc, qs, and qi.
>             rh_hack= rh(k)
> -           !ensure adequate RH & q1 when qi is at least 1e-9
> -           if (qi(k)>1.e-9) then
> -              rh_hack =min(1.0_kind_phys, rhcrit + 0.07*(9.0 + log10(qi(k))))
> +           !ensure adequate RH & q1 when qi is at least 1e-9 (above the PBLH)
> +           if (qi(k)>1.e-9 .and. zagl .gt. pblh2) then
>
> I don't think there is anything wrong with that line of code. This just
> seems to be (one of many) tweaks that can apparently push the model over
> the edge. You aren't suggesting this is actually the culprit, right?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/ufs-community/ccpp-physics/pull/120#discussion_r1387309169>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AGCO5UDFW7TH3EBB5FYJ4B3YDQIXPAVCNFSM6AAAAAA6CZ25KCVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMYTOMRRGQ2DIMBWHA>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

dustinswales commented:
@gthompsnWRF I think you're correct. These lines in progcld_thompson should be consistent w/ what is done when calling calc_effectRad, https://github.com/ufs-community/ccpp-physics/blob/ufs/dev/physics/GFS_rrtmg_pre.F90#L899.

grantfirl commented:
> @AndersJensen-NOAA and @joeolson42 Is RRFS using `subroutine progcld_thompson` within `radiation_clouds.f`?
> 
> I am asking because I have concerns that the cloud fractions between zero and one might be given the value of `re_qc_min` (2.5 microns) versus what I assume for the "fake clouds" which is found in these lines of code . . .
> 
> ```
>             if (qc1d(k).gt.clwmin .and. cldfra1d(k).lt.ovcst) then
>                cwp(i,k) = qc1d(k) * dz1d(k)*1000.
>                if ((xland-1.5).GT.0.) then                               !--- Ocean
>                   rew(i,k) = 9.5
>                else                                                      !--- Land
>                   rew(i,k) = 5.5
>                endif
>             endif
> ```
> 
> So I am assuming rather large droplets in the fractional clouds for which there could be zero grid-scale microphysics for which to compute the mathematical value of cloud water effective radii. In that case, the subroutine within module_mp_thompson to compute effective radii would return `re_qc_min` (or 2.5 microns), but it can be overridden here in the case with cloud fraction less than 1.0 (but greater than zero) so the clouds won't be so bright. Also, in your PDF, what are you meaning by clouds are "too bright?" If you're looking at things relative to some pseudo satellite operator in the visible spectra, then who has done the checks for consistency with the operator calculating albedo or whatever? Those include tables for single-scattering albedo and asymmetry factor and so on.
> 
> Also tagging @RuiyuSun and @yangfanglin

At least for the RRFS regression tests, I see that icloud=0 and therefore prgclduni is called. I don't know if there are other circumstances where RRFS might use icloud=3 and call prgcld_thompson.

joeolson42 commented:
@gthompsnWRF @AndersJensen-NOAA @grantfirl Sorry to be slow to respond, but yes, we use progclduni.

grantfirl committed:
Merge branch 'ufs/dev' into mynnthomp_oct2023

grantfirl committed:
add extra condition in line 2655 of module_sf_noahmp_glacier.F90 to avoid FPE errors in some tests

grantfirl commented on physics/module_sf_noahmp_glacier.F90:
```diff
@@ -2652,7 +2652,7 @@ subroutine snowwater_glacier (nsnow   ,nsoil   ,imelt    ,dt     ,sfctmp , & !in
 !to obtain equilibrium state of snow in glacier region
        
-   if(sneqv > mwd) then   ! 100 mm -> maximum water depth
+   if(sneqv > mwd .and. isnow /= 0) then   ! 100 mm -> maximum water depth
```
@barlage @HelinWei-NOAA Are you both OK with adding this to avoid a floating point exception that is sometimes triggered (e.g. in the rrfs_v1beta_debug_intel test for this PR and in previous PRs: https://github.com/ufs-community/ccpp-physics/issues/114)?

barlage replied:
@grantfirl yes, I'm good with this change. I was going to suggest bumping the mwd up to a larger number again, but this should fix the issue. It's still puzzling to me as to why this happens. Will look into it more to track down the root cause.

gthompsnWRF commented:
> @gthompsnWRF @AndersJensen-NOAA @grantfirl Sorry to be slow to respond, but yes, we use progclduni.

**EDITED!  See 2nd paragraph**

This is all wrong.  `progclduni` uses the GFDL microphysics radii for cloud water and ice.  This is all wrong, especially for Thompson microphysics.  I have a slide deck showing the problems of GFDL's assignment of ice and water sizes.  I thought the pre-calculated Thompson MP radii were carried into this subroutine if not module.

Actually, this line of code
```
      if (effr_in) then
```
gets the drop and ice sizes from the existing arrays computed by gthompsn's method.  However, since the fractional clouds have no cloud water or ice, they will default to the smallest sizes.  Hence, what I'm pointing out is that the assumed minimum size of 2.5 microns is being used for the fractional clouds away from places with grid-explicit cloud water.

So as I typed above, the chunk of code in `progcld_thompson` could be used as a new addition to `progclduni` to explicitly handle drop/ice sizes in the cloud fraction, non-grid-explicit clouds.  This _may help_ alleviate with Anders/Joe are seeing as "too bright" clouds without dramatically affecting the GFS results.

joeolson42 commented:
@gthompsnWRF @AndersJensen-NOAA @grantfirl 

No thanks for bringing this up on the final days of the PR, but I think this is worth looking into for a follow-up PR. This will require more testing an evaluation.

I never saw any 2.5 micron effective radii over water. It was more like 3.5 to 6 range, but a lot of them. Anyway, let's handle this off line. Anders and I are willing to look into this very soon. I'm marking this as resolved for this PR.

yangfanglin commented:
Ruiyu and Anning,Could you please take a look at this.  I am confident when Thompson MP is called in gfs/gefs, the corresponding radii from Thompson NO are used for radiation calculation.

RuiyuSun commented:
@yangfanglin Yes. In the GFS, when Thomspon MP is used, the radii from the Thompson MP are used in the radiation calculation.

yangfanglin commented:
Hi, Fanglin:

Effective radius (e.g., re_cloud) is calculated in calc_effectRad
of module_mp_thompson.F90, passed into  progcld_thompson (e.g., rew
assigned to cld_reliq) of radiation_clouds.f. cld_reliq etc. are used for
radiation calculation  in rrtmgp_lw_main.F90 and rrtmgp_sw_main.F90.

Anning

On Thu, Nov 9, 2023 at 4:14 PM Fanglin Yang - NOAA Federal <
***@***.***> wrote:

> Ruiyu and Anning,
>
> Could you please take a look at this.  I am confident when Thompson MP is
> called in gfs/gefs, the corresponding radii from Thompson NO are used for
> radiation calculation.
>
> Fanglin
>
> Sent from my iPhone
>
> On Nov 9, 2023, at 10:30 AM, gthompsnWRF ***@***.***> wrote:
>
> ﻿
>
> @gthompsnWRF <https://github.com/gthompsnWRF> @AndersJensen-NOAA
> <https://github.com/AndersJensen-NOAA> @grantfirl
> <https://github.com/grantfirl> Sorry to be slow to respond, but yes, we
> use progclduni.
>
> This is all wrong. progclduni uses the GFDL microphysics radii for cloud
> water and ice. This is all wrong, especially for Thompson microphysics. I
> have a slide deck showing the problems of GFDL's assignment of ice and
> water sizes. I thought the pre-calculated Thompson MP radii were carried
> into this subroutine if not module.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/ufs-community/ccpp-physics/pull/120#issuecomment-1804343899>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AKY5N2K7UYTYKM652RFMRDDYDUOLLAVCNFSM6AAAAAA6CZ25KCVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMYTQMBUGM2DGOBZHE>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>
>

yangfanglin commented:
The so-called "progclduni" formally designed for MG2/3 make the code
difficult to read, but the principle is the same. Greg used icloud=3 to
make the code easy to read though.

On Thu, Nov 9, 2023 at 5:54 PM Anning Cheng - NOAA Affiliate <
***@***.***> wrote:

> Hi, Fanglin:
>
> Effective radius (e.g., re_cloud) is calculated in calc_effectRad
> of module_mp_thompson.F90, passed into  progcld_thompson (e.g., rew
> assigned to cld_reliq) of radiation_clouds.f. cld_reliq etc. are used for
> radiation calculation  in rrtmgp_lw_main.F90 and rrtmgp_sw_main.F90.
>
> Anning
>
> On Thu, Nov 9, 2023 at 4:14 PM Fanglin Yang - NOAA Federal <
> ***@***.***> wrote:
>
>> Ruiyu and Anning,
>>
>> Could you please take a look at this.  I am confident when Thompson MP is
>> called in gfs/gefs, the corresponding radii from Thompson NO are used for
>> radiation calculation.
>>
>> Fanglin
>>
>> Sent from my iPhone
>>
>> On Nov 9, 2023, at 10:30 AM, gthompsnWRF ***@***.***>
>> wrote:
>>
>> ﻿
>>
>> @gthompsnWRF <https://github.com/gthompsnWRF> @AndersJensen-NOAA
>> <https://github.com/AndersJensen-NOAA> @grantfirl
>> <https://github.com/grantfirl> Sorry to be slow to respond, but yes, we
>> use progclduni.
>>
>> This is all wrong. progclduni uses the GFDL microphysics radii for cloud
>> water and ice. This is all wrong, especially for Thompson microphysics. I
>> have a slide deck showing the problems of GFDL's assignment of ice and
>> water sizes. I thought the pre-calculated Thompson MP radii were carried
>> into this subroutine if not module.
>>
>> —
>> Reply to this email directly, view it on GitHub
>> <https://github.com/ufs-community/ccpp-physics/pull/120#issuecomment-1804343899>,
>> or unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/AKY5N2K7UYTYKM652RFMRDDYDUOLLAVCNFSM6AAAAAA6CZ25KCVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMYTQMBUGM2DGOBZHE>
>> .
>> You are receiving this because you were mentioned.Message ID:
>> ***@***.***>
>>
>>

tanyasmirnova committed:
Computation of surface fire heat for use in LSM to rrfs_smoke_wrapper.
Also, added the fraction of grid cell that is burned out.

tanyasmirnova committed:
Small changes to fire feedback to RUC LSM.
One more variable is added: fraction of grid cell burned by the fire.
This fraction is used to take into account fire's effect on surface albedo.
Also, added some debug prints.
These changes will change the results only when fire feedback is turned on. The default is .false.

dustinswales commented:
> Hi, Fanglin: Effective radius (e.g., re_cloud) is calculated in calc_effectRad of module_mp_thompson.F90, passed into progcld_thompson (e.g., rew assigned to cld_reliq) of radiation_clouds.f. cld_reliq etc. are used for radiation calculation in rrtmgp_lw_main.F90 and rrtmgp_sw_main.F90. Anning On Thu, Nov 9, 2023 at 4:14 PM Fanglin Yang - NOAA Federal < ***@***.***> wrote:
> […](#)
> Ruiyu and Anning, Could you please take a look at this. I am confident when Thompson MP is called in gfs/gefs, the corresponding radii from Thompson NO are used for radiation calculation. Fanglin Sent from my iPhone On Nov 9, 2023, at 10:30 AM, gthompsnWRF ***@***.***> wrote: ﻿ @gthompsnWRF <https://github.com/gthompsnWRF> @AndersJensen-NOAA <https://github.com/AndersJensen-NOAA> @grantfirl <https://github.com/grantfirl> Sorry to be slow to respond, but yes, we use progclduni. This is all wrong. progclduni uses the GFDL microphysics radii for cloud water and ice. This is all wrong, especially for Thompson microphysics. I have a slide deck showing the problems of GFDL's assignment of ice and water sizes. I thought the pre-calculated Thompson MP radii were carried into this subroutine if not module. — Reply to this email directly, view it on GitHub <[#120 (comment)](https://github.com/ufs-community/ccpp-physics/pull/120#issuecomment-1804343899)>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AKY5N2K7UYTYKM652RFMRDDYDUOLLAVCNFSM6AAAAAA6CZ25KCVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMYTQMBUGM2DGOBZHE> . You are receiving this because you were mentioned.Message ID: ***@***.***>

I think you mean radlw_main and radsw_main?
Let's not drag RRTMGP into this, which treats radiatively active clouds differently. Notice I used "radiatively active clouds" not "fake clouds", they are very real to the radiation

joeolson42 commented:
I agree about not dragging anything else into this PR. Too much has already been dragged in.

grantfirl committed:
Merge branch 'ruclsm_lai' into mynnthomp_oct2023

grantfirl committed:
fix intent in mp_thompson.meta

grantfirl commented:
@dustinswales Please re-review when you get a chance since you requested changes.

grantfirl committed:
Merge branch 'ufs/dev' into mynnthomp_oct2023

jkbk2004 commented:
All tests are done at https://github.com/ufs-community/ufs-weather-model/pull/1945. @grantfirl @dustinswales Please, go ahead to merge this pr.
